File: src/main/java/com/laamella/code_state_machine/action/FinishableAction.java
package com.laamella.code_state_machine.action;

import com.laamella.code_state_machine.Action;
import com.laamella.code_state_machine.Precondition;

/**
 * An action which finishes at some time in the future. A transition can wait
 * for the action to be finished by using the isFinished precondition.
 */
public interface FinishableAction<E> extends Action {
	Precondition<E> isFinished();
}

File: scala/src/main/scala/com/laamella/code_state_machine/action/FinishableAction.scala
package com.laamella.code_state_machine.action

import com.laamella.code_state_machine.Condition

/**
 * An action which finishes at some time in the future. A transition can wait
 * for the action to be finished by using the finished condition.
 */
trait FinishableAction[Event] extends (() => Unit) {
	def finished: Condition[Event]
}
